### ISSUE TITLE
[AniDUB] returns lots of unrelated results

### ISSUE THREAD

**Issue by janot** [Jackett/Jackett#8962]
![Capture](https://user-images.githubusercontent.com/1689966/84513396-067a2800-acd2-11ea-84ee-0e70915ddbdf.PNG)

Site had Advanced search option, which returns normal results (nothing in this case):
![Capture](https://user-images.githubusercontent.com/1689966/84513736-8c966e80-acd2-11ea-85bd-ded51ac8787d.PNG)

This problem was already mentioned https://github.com/Jackett/Jackett/issues/5399#issuecomment-502343029 by @MsKrypt1 
I think it should be fixed if possible

**Comment by ngosang** [comment]
@MsKrypt1 @dimqua could you help?

**Comment by kekal** [comment]
@janot @dm3ch 

I got a version with these changes to use it with Sonarr (with docker) and concluded that after this fix, the search actually doesn't work at all.
Sonarr generates queries in such a way that Jackett always returns 0 for the Anidub indexer.

Please, if anyone is using the same tandem, confirm if I'm not the only one with this problem.
Thanks


PS Am I understand right that any titleonly value less than 3 generates many unrelated query results?

**Comment by dm3ch** [comment]
@kekal 
I just tried most recent version in master branch (just Jackett) (I checked that it contains my fix) and all worked fine for me.
I tried to with Naruto title (and Наруто russian tittle). I also tested some other.

About Jackett + Sonarr I have no idea what's going on. Maybe I would have some time for further debug on next weekends

**Comment by ilike2burnthing** [comment]
Fix should be out in ~4hrs

**Comment by kekal** [comment]
> @kekal
> I just tried most recent version in master branch (just Jackett) (I checked that it contains my fix) and all worked fine for me.
> I tried to with Naruto title (and Наруто russian tittle). I also tested some other.
> 
> About Jackett + Sonarr I have no idea what's going on. Maybe I would have some time for further debug on next weekends

I still don't understand why, but after the 2438d5425294644770712eea31843db47e122b32, the search results became correct through Sonarr.
Thank you very much.

**Comment by ilike2burnthing** [comment]
Any search with more than one keyword wasn't working, as rather than searching for. e.g. `720p 2020` it was searching for `720p+2020`, which the site sees as a single word, rather than `720p` AND `2020`.

**Comment by kekal** [comment]
@dm3ch 
Excuse me, but my successful testing was optimistic.
It's reproducible not on all titles.

I've made a detailed reproduce with logs.

### So, the situation **before these changes**:
1. Sonarr generates three queries:
    - 20-12-31 12:17:15.1|Debug|Torznab|Downloading Feed http://172.17.0.1:9117/api/v2.0/indexers/anidub/results/torznab/api?t=search&cat=5000,5070,8000,100002,100003,100004,100005,100010,100011,100013,100014&extended=1&apikey=(removed)&offset=0&limit=100&q=Maou%20jou%20de%20Oyasumi+12
    - 20-12-31 12:17:21.2|Debug|Torznab|Downloading Feed http://172.17.0.1:9117/api/v2.0/indexers/anidub/results/torznab/api?t=search&cat=5000,5070,8000,100002,100003,100004,100005,100010,100011,100013,100014&extended=1&apikey=(removed)&offset=0&limit=100&q=Maoujou%20de%20Oyasum+12
    - 20-12-31 12:17:31.5|Debug|Torznab|Downloading Feed http://172.17.0.1:9117/api/v2.0/indexers/anidub/results/torznab/api?t=search&cat=5000,5070,8000,100002,100003,100004,100005,100010,100011,100013,100014&extended=1&apikey=(removed)&offset=0&limit=100&q=Sleepy%20Princess%20in%20the%20Demon%20Castle+12

2. Jackett reaction (I don't know, how to create a detailed log):
```
2020-12-31 12:16:34.0446 Info Jackett startup finished in 26.555 s 
2020-12-31 12:17:04.5974 Info Found 15 releases from AniDUB 
2020-12-31 12:17:21.2218 Info Found 13 releases from AniDUB for: Maou jou de Oyasumi 12 
2020-12-31 12:17:31.4958 Info Found 1 releases from AniDUB for: Maoujou de Oyasum 12 
2020-12-31 12:17:48.0152 Info Found 41 releases from AniDUB for: Sleepy Princess in the Demon Castle 12
```
3. Actual releases fetched by Sonarr:
    - One proper result:
*Maoujou de Oyasumi S01E01-E12 [RUS] [HDTV 1080p]*
    - ...and bunch of incorrect results, that were actually fixed here:
```MAOU GAKUIN NO FUTEKIGOUSHA S01E01-E13 [RUS] [HDTV 1080p]
Uchi no Ko no Tame naraba, Ore wa Moshikashitara Maou mo Taoseru kamo Shirenai S01E01-E12 [RUS] [HDTV 1080p]
Isekai Maou to Shoukan Shoujo no Dorei Majutsu S01E01-E12 [RUS] [PSP]
Isekai Maou to Shoukan Shoujo no Dorei Majutsu S01E01-E12 [RUS] [SDTV]
Isekai Maou to Shoukan Shoujo no Dorei Majutsu S01E01-E12 [RUS] [BDRip 1080p]
Isekai Maou to Shoukan Shoujo no Dorei Majutsu S01E01-E12 [RUS] [HDTV 720p]
Hataraku Maou-sama! S01E01-E13 [RUS] [PSP]
Hataraku Maou-sama! S01E01-E13 [RUS] [SDTV]
Hataraku Maou-sama! S01E01-E13 [RUS] [HDTV 720p]
Maoyuu Maou Yuusha S01E01-E12 [RUS] [PSP]
Maoyuu Maou Yuusha S01E01-E12 [RUS] [SDTV]
Maoyuu Maou Yuusha S01E01-E12 [RUS] [BDRip 720p]
Kaibutsu Oujo S01E01-E25 [RUS] [PSP]
Kaibutsu Oujo S01E01-E25 [RUS] [SDTV]
Seiken no Blacksmith S01E01-E12 [RUS] [SDTV]
Seiken no Blacksmith S01E01-E12 [RUS] [BDRip 1080p]
Seiken no Blacksmith S01E01-E12 [RUS] [HDTV 720p]
Slap Up Party: Arad Senki S01E01-E26 [RUS] [SDTV]
Slap Up Party: Arad Senki S01E01-E26 [RUS] [HDTV 720p]
Kikoushi-Enma (OVA) [RUS] [DVD480]
Lupin Sansei: Princess of the Breeze [RUS] [PSP]
Lupin Sansei: Princess of the Breeze [RUS] [SDTV]
Lupin Sansei: Princess of the Breeze [RUS] [HDTV 720p]
Yondemasu yo, Azazel-san S01E01-E13 [RUS] [PSP]
Yondemasu yo, Azazel-san S01E01-E13 [RUS] [SDTV]
Yondemasu yo, Azazel-san S01E01-E13 [RUS] [BD480]
Princess Principal S01E01-E12 [RUS] [PSP]
Princess Principal S01E01-E12 [RUS] [SDTV]
Princess Principal S01E01-E12 [RUS] [HDTV 720p]
Beelzebub. S01E01-E60 [RUS] [PSP]
Beelzebub. S01E01-E60 [RUS] [SDTV]
Beelzebub. S01E01-E60 [RUS] [HDTV 720p]
Higurashi no Naku Koro ni S01E01-E26 [RUS] [PSP]
Higurashi no Naku Koro ni S01E01-E26 [RUS] [SDTV]
Higurashi no Naku Koro ni S01E01-E26 [RUS] [HDTV 720p]
The Soultaker S01E01-E13 [RUS] [PSP]
The Soultaker S01E01-E13 [RUS] [SDTV]
The Soultaker S01E01-E13 [RUS] [DVD480]
Kaibutsu Oujo OVA S01E01-E3 [RUS] [PSP]
Kaibutsu Oujo OVA S01E01-E3 [RUS] [SDTV]
Kaibutsu Oujo OVA S01E01-E3 [RUS] [BDRip 720p]
Dog Days S01E01-E13 [RUS] [PSP]
Dog Days S01E01-E13 [RUS] [BDRip 720p]
Dog Days S01E01-E13 [RUS] [BDRip 1080p]
Dog Days S01E01-E13 [RUS] [HDTV 720p]
Seiden RG Veda S01E01-E2 [RUS] [DVD480]
Sengoku Basara Ni S01E01-E12 [RUS] [PSP]
Sengoku Basara Ni S01E01-E12 [RUS] [SDTV]
Sengoku Basara Ni S01E01-E12 [RUS] [HDTV 720p]
Soul Eater S01E01-E51 [RUS] [PSP]
Soul Eater S01E01-E51 [RUS] [SDTV]
Soul Eater S01E01-E51 [RUS] [BDRip 720p]
Soul Eater S01E01-E51 [RUS] [HDTV 720p]
```


### The situation **after these changes**:
1. Sonarr generates three same queries:
    - 20-12-31 12:26:46.9|Debug|Torznab|Downloading Feed http://172.17.0.1:9117/api/v2.0/indexers/anidub/results/torznab/api?t=search&cat=5000,5070,8000,100002,100003,100004,100005,100010,100011,100013,100014&extended=1&apikey=(removed)&offset=0&limit=100&q=Maou%20jou%20de%20Oyasumi+12
    - 20-12-31 12:26:51.3|Debug|Torznab|Downloading Feed http://172.17.0.1:9117/api/v2.0/indexers/anidub/results/torznab/api?t=search&cat=5000,5070,8000,100002,100003,100004,100005,100010,100011,100013,100014&extended=1&apikey=(removed)&offset=0&limit=100&q=Maoujou%20de%20Oyasum+12
    - 20-12-31 12:27:03.1|Debug|Torznab|Downloading Feed http://172.17.0.1:9117/api/v2.0/indexers/anidub/results/torznab/api?t=search&cat=5000,5070,8000,100002,100003,100004,100005,100010,100011,100013,100014&extended=1&apikey=(removed)&offset=0&limit=100&q=Sleepy%20Princess%20in%20the%20Demon%20Castle+12

2. Jackett reaction is empty:
```
2020-12-31 12:26:33.6323 Info Jackett startup finished in 10.506 s 
2020-12-31 12:26:51.3765 Info Torznab search in AniDUB for Maou jou de Oyasumi 12 => Found 0 releases 
2020-12-31 12:27:03.1176 Info Torznab search in AniDUB for Maoujou de Oyasum 12 => Found 0 releases 
2020-12-31 12:27:15.1091 Info Torznab search in AniDUB for Sleepy Princess in the Demon Castle 12 => Found 0 releases 
```


### Additional information:
If I change the query from **Maou%20jou%20de%20Oyasumi**+12 to **Maou%20jou%20de+12**, Jackett returns correct results.

**Comment by ilike2burnthing** [comment]
https://tr.anidub.com/index.php?do=search > click `Расширенный поиск` > use those 3 search terms > click `Искать в статьях` and select `Искать в заголовках статей` instead > perform those searches again

That's the difference. So I'm assuming we could revert back to the original and apply an `andmatch` filter to the results? Not sure how to do that for C# indexers though. @ngosang or @dm3ch may have a better idea.

**Comment by dm3ch** [comment]
I'll take a look on 1th or 2nd January

**Comment by ilike2burnthing** [comment]
@dm3ch ping

**Comment by dm3ch** [comment]
@ilike2burnthing Sorry, January was too busy month for me.

I have tried to implement fix the way we discussed above, but filtering seems to not work properly. :(
https://github.com/Jackett/Jackett/pull/10972

Interesting, that this type of filtering worked well in my Anilibria indexer, if i right remember.

**Comment by dm3ch** [comment]
Or maybe I have tested my PR wrong. So I would be happy if someone could help with current #10972 test

**Comment by kekal** [comment]
I would like to help with testing, but my setup made with docker.
How can I deploy a custom branch in this case? Is there some way to use artefacts from this PR from [Azure](https://dev.azure.com/Jackett/Jackett/_build/results?buildId=2791&view=results)?
Or maybe I can just easily build it for myself?

**Comment by kekal** [comment]
Update. I managed to build [container](https://github.com/linuxserver/docker-jackett/blob/master/Dockerfile) with   [binary from this PR](https://dev.azure.com/Jackett/Jackett/_build/results?buildId=2791&view=artifacts&pathAsName=false&type=publishedArtifacts).
But running the container, I cannot even reach the Jackett webpage. So, there are some details that I cannot see from in the [CI file.](https://github.com/linuxserver/docker-jackett/blob/master/Jenkinsfile#L361).
Сan't move on.

**Comment by ilike2burnthing** [comment]
Looks like it's working here.

**On site:**
`Maou jou de Oyasumi 12` returns 6 results, 1 relevant
`Maoujou de Oyasum 12` returns 1 relevant result
`Sleepy Princess in the Demon Castle 12` returns 15 results (on page 1), 0 relevant

**In Jackett:**
`Maou jou de Oyasumi 12` returns 1 relevant result
`Maoujou de Oyasum 12` returns 1 relevant result
`Sleepy Princess in the Demon Castle 12` returns 0 results

**Comment by ilike2burnthing** [comment]
https://github.com/Jackett/Jackett/pull/10972

**Comment by kekal** [comment]
Indexer works fine for me.

As I see this fix does not make similar to the #10931.
It is a pity.
